uhubctl.c

```diff
@@ -919,16 +919,16 @@ int main(int argc, char *argv[])
             printf("\n");
             break;
         case 'l':
-            strncpy(opt_location, optarg, sizeof(opt_location));
+            strncpy(opt_location, optarg, sizeof(opt_location) - 1);
```

Comment:
Unfortunately, this still does not guarantee 100% reliable fix for undefined behavior - `opt_location` could still have contained non-zero characters at the very end of the buffer.
Ideally, we'd want to use `slrlcpy`, but it is not standard functon.
Can you please replace it with following `snprintf` snippet? Googling around snprintf seems to be always adding zero terminator. I tested it locally, it works fine and passes your test:
```
-            strncpy(opt_location, optarg, sizeof(opt_location));
+            snprintf(opt_location, sizeof(opt_location), "%s", optarg);
```
Using `snprintf(opt_location, sizeof(opt_location), optarg)` works too, but it emits lots of warnings, so it is better to use `"%s"` format string.

Reply:
`opt_location` is initialized as 32 null characters by https://github.com/real-or-random/uhubctl/blob/202208-strncpy/uhubctl.c#L214: 
```
static char opt_location[32] = "";
```
So if `snprintf` writes at most 31, it won't overwrite the last null character.

But I can still switch to `snprintf` if you think it's better. Shall I?

Reply:
Using `snprintf` will work regardless of `opt_location` (or any other string parameter) being initialized or not.
So if we want to 100% avoid any possible UB, that should be way to go.

I have seen people to copy/paste this code as part of their libraries (even if this is against GPL), or they were adding external repeat loops around `main()` so they still had some static variables not properly initialized upon second loop (don't ask me why!).

Alternatively we can add a call to `bzero(opt_location, sizeof(opt_location))` as a nuclear option :-)

Reply:
you're right, it's cleaner and less coupled. updated.